src/cargo/ops/fix.rs

```diff
@@ -471,4 +501,16 @@ impl FixArgs {
 
         Ok(())
     }
+
+    fn next_edition(&self) -> &str {
+        match self.enabled_edition.as_ref().map(|s| &**s) {
+            // 2015 -> 2018,
+            None | Some("2015") => "2018",
+
+            // This'll probably be wrong in 2020, but that's future Cargo's
+            // problem. Eventually though we'll just add more editions here as
+            // necessary.
+            _ => "2018",
```

Comment:
Hmmmm. Is it actually only future cargo's problem? What if I run `cargo fix --edition` on an 2018-edition project _right now_? I'd naively assume this method to deal with that. As it stands right now it sometimes returns a string we can't really work with, and instead assumes we call a function like `verify_not_preparing_for_enabled_edition`.

Maybe it's overkill, but I'd have this return a `Result<&str, CantUpgradeEditionError>`, with `enum CantUpgradeEditionError { UnknownEdition, LatestKnownEdition }`.

Reply:
Yeah I was wondering this as well, and it's true that the natural interperetation of this flag on the 2018 edition is to avoid doing anything. Currently though we have a good property where if you mix up the transition steps you'll get warnings and errors to nudge you in the right direction. If we take this approach, though, if you enable the edition to soon you won't get a helpful error but rather a deluge of resolution errors (due to `crate::`). 

For that reason I figured it'd be best to stick to a policy of "`--edition` can't be used when you're already on the latest edition" and that way we should hopefully not surprise too many people and otherwise continue to help nudge towards the new edition along the prescribed path.